pkg/blockdevice/lvm.go

```diff
@@ -210,19 +216,65 @@ func (l *Lvm) Deactivate(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
+// GetDevice generates the name used in /dev/mapper for the vg/lv
 func (l *Lvm) GetDevice() string {
+	// Add a hypen to our name to match up with /dev/mapper.
+	//
+	// According to this article, https://access.redhat.com/solutions/656673
+	// the hyphen '-' is used to generate a unique single string to reference the device.
+	// The hyphen is used by LVM2 as the field separator when constructing device-mapper device names
+	// from LVM2 volume group and logical volume names:
+	//		dm name := 'vg_name' + '-' + 'lv_name'
+	// Since LVM2 permits the hyphen as a character within a VG or LV name this scheme requires embedded
+	// hyphens to be escaped when constructing the dm name. The escaping scheme is to double any embedded hyphens so for e.g.:
+	//      Volume Group: my-vg
+	//      Logical Volume: my-lv
+	// Is encoded as: my--vg-my--lv
+	// NOTE!!!
+	// The encoding method might be changed without notice in a future release, so you should always use /dev/vg/lv to access a device, and dmsetup splitname to decode a name.
+
 	return fmt.Sprintf("/dev/mapper/%s-%s", strings.Replace(l.VolumeGroup.Name, "-", "--", -1), strings.Replace(l.LogicalVolume.Name, "-", "--", -1))
 }
 
+// CheckFormatted determines if the device has been formatted
 func (l *Lvm) CheckFormatted() (bool, error) {
-	output, err := command.Run(fmt.Sprintf("wipefs --noheadings --output type %s", l.GetDevice()), l.Log)
-	if err != nil {
-		return false, fmt.Errorf("could not run wipefs to determine if device is formatted: %w", err)
+
+	// Default to 10 second timeout
+	retryPeriod := 10 * time.Second
+
+	// Look for environment variable to override
+	timeoutString, found := os.LookupEnv("NNF_WIPEFS_MAPPER_WAIT_COMMAND_TIMEOUT")
+	if found {
+		timeout, err := strconv.Atoi(timeoutString)
+		if err == nil {
+			if timeout > 0 {
+				retryPeriod = time.Duration(timeout) * time.Second
+			}
+		}
 	}
 
-	if len(output) == 0 {
-		return false, nil
+	// Give the device some time to appear in /dev/mapper
+	var err error
+	var output string
+
+	// The /dev/mapper device can take a while.
+	// Retry failing wipefs to give it time.
+	device := l.GetDevice()
+	for start := time.Now(); time.Since(start) < retryPeriod; time.Sleep(time.Second) {
+
+		output, err = command.Run(fmt.Sprintf("wipefs --noheadings --output type %s", device), l.Log)
```

Comment:
I'd consider adding a `stat()` call before `command.Run()` to verify the path exists and we should expect `wipefs` to succeed.

Reply:
I considered that and in fact coded it. It seems we need to poll on some command, be it `stat` or `wipefs`. Is there a benefit to waiting with `stat()` above waiting with `wipefs`? In either case, if the path we are waiting for fails to show up, we are left with an error.

Reply:
Tony, you can use `strings.Contains(err, "mumble")` to verify that wipefs failed because the device isn't present.  Then "mumble" would be just a snippet from the error, such as "not found".  Any other error and maybe we shouldn't be looping.

As it's coded now, if the first error in the loop and the last error in the loop are different, and both have nothing to do with "not found", then we'll never know the first error.  We should probably know that first error.

Reply:
The advantage of checking if the files exists first is you can do it with a system call.  In the error case this entirely avoids the fork/exec and any execution `wipefs` might need to do before it fails.  It looks like both the [lstat](https://pkg.go.dev/syscall#Lstat) and [access](https://pkg.go.dev/syscall#Access) syscalls are available.  Since you just need to check for existence, `access(2)` is probably the easiest way to go.

If you're feeling really ambitious `inotify` is the way to handle this on Linux without polling.  If I understand correctly [fsnotify](https://pkg.go.dev/github.com/fsnotify/fsnotify) is the preferred portable wrapper for this in go, but I'm not sure offhand what additional dependencies you might need for that.